selfservice/strategy/oidc/provider.go

```diff
@@ -13,28 +15,44 @@ type Provider interface {
 	AuthCodeURLOptions(r ider) []oauth2.AuthCodeOption
 }
 
+// ConvertibleBoolean can unmarshal both booleans and strings.
+// This is needed as Apple casually sends the email_verified field as a string.
+type ConvertibleBoolean bool
```

Comment:
not sure if to add this type here next to where it's used or in a separate file

Reply:
Looks like general purpose type. I'm slightly leaning towards putting it into some commonly accessible package.

Reply:
Like put it in https://github.com/Wikia/kratos/tree/master/x?

Reply:
Yeah. I guess.